x-pack/plugins/observability_ai_assistant/public/components/insight/actions_menu.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState } from 'react';
+import { EuiButtonIcon, EuiContextMenu, EuiPopover } from '@elastic/eui';
+import { UseGenAIConnectorsResult } from '../../hooks/use_genai_connectors';
+
+export function ActionsMenu({
+  connectors,
+  onEditPrompt,
+}: {
+  connectors: UseGenAIConnectorsResult;
+  onEditPrompt: () => void;
+}) {
+  const [isPopoverOpen, setPopover] = useState(false);
+
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setPopover(false);
+  };
+
+  const panels = [
+    {
+      id: 0,
+      title: 'Actions',
+      items: [
+        {
+          name: `Connectors: ${connectors.selectedConnector}`,
+          icon: 'wrench',
+          panel: 1,
+        },
+        {
+          name: 'Edit prompt',
+          icon: 'documentEdit',
+          onClick: () => {
+            onEditPrompt();
+            closePopover();
+          },
+        },
+      ],
+    },
+    {
+      id: 1,
+      title: 'Connectors',
+      items: connectors.connectors?.map((connector) => {
+        return {
+          name: connector.name,
+          icon: connector.id === connectors.selectedConnector ? 'check' : undefined,
+          onClick: () => {
+            connectors.selectConnector(connector.id);
+            closePopover();
+          },
+        };
+      }),
+    },
+  ];
+
+  const button = (
+    <EuiButtonIcon
+      data-test-subj="observabilityAiAssistantInsightActionsButtonIcon"
```

Comment:
`EuiButtonIcon` needs an `aria-label` in order to not get a warning in the console

Reply:
fixed in https://github.com/elastic/kibana/pull/176086/commits/08c86cbb61a9e997a28bef608c11f657cdb74e50